math.p8

```diff
@@ -73,35 +73,21 @@ end
 
 
 ------------------------------------------------------------------------
--- distance functions partially thanks to https://www.lexaloffle.com/bbs/?pid=119363
-
--- distance calculation
--- uses dist_naive for small values, dist_trig for large
--- tokens: 46
-function dist(x,y)
- if (abs(x) < 128 and abs(y) < 128) return dist_naive(x,y)
- return dist_trig(x,y)
-end
-
--- naive distance calculation
--- tokens: 15
--- cycles: 9 + 28 = 37 (lua+sys)
--- numerical issues: breaks down for dist*dist>=32768, can cause big problems
-function dist_naive(x,y)
- return sqrt(x*x+y*y)
+-- distance between two points
+-- copied from vector.p8
+local function dist(x1, y1, x2, y2)
+ local x, y = abs(x2 - x1), abs(y2 - y1)
+ if (x < 128 and y < 128) return sqrt(x * x + y * y) -- remove[25,54] slight decrease in accuracy
+ local d = max(x, y)
+ local n = x / d * y / d
+ return sqrt(n * n + 1) * d
 end
 
--- distance calculation using trig
--- tokens: 23
--- cycles: 21 + 0 = 21 (lua+sys)
--- error in range (0,0)-(127,127): max 0.0017, avg 0.0004
-function dist_trig(x,y)
- local a=atan2(x,y)
- return x*cos(a)+y*sin(a)
-end
 
--- distance squared
-local function dist_squared(x, y)
+------------------------------------------------------------------------
+-- distance squared between two points
+local function distsqr(x1, y1, x2, y2)
+ local x, y = abs(x2 - x1), abs(y2 - y1)
  return x * x + y * y
 end
 
```

Comment:
Changes above this feel like they might be a rebase from older code? Please look over this section and confirm you like these changes before we discuss them.

Reply:
I'm actually not sure what happened here. I pulled latest from master when I merged... perhaps I forgot to git fetch before merging. Could you link to the latest so I can cross reference?

Reply:
As of the push I just made, which conflicts a bunch more of your stuff (sorry), https://github.com/sparr/pico8lib/tree/41382b16bafc2111fc9415d80c818cc9f25345b7 is the current commit. I anticipate leaving the repo alone until this PR is merged.

Reply:
Thanks, I'll see what I can do.